packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -70,31 +75,47 @@ export class WaferMap extends FoundationElement {
 
     private renderQueued = false;
     private dataManager: DataManager | undefined;
-
+    private renderer: RenderingModule | undefined;
+    private resizeObserver: ResizeObserver | undefined;
     public override connectedCallback(): void {
         super.connectedCallback();
+        this.resizeObserver = new ResizeObserver(entries => {
+            const entry = entries[0];
+            if (entry === undefined) return;
+            const { height, width } = entry.contentRect;
+            this.canvasSideLength = Math.min(height, width);
+        });
+        this.resizeObserver.observe(this);
         this.queueRender();
     }
 
+    public override disconnectedCallback(): void {
+        this.resizeObserver?.unobserve(this);
+    }
+
     /**
      * @internal
      */
     public render(): void {
         this.renderQueued = false;
+        if (this.canvasSideLength === undefined || this.canvasSideLength === 0) return;
+        this.renderer?.clearCanvas(
+            this.canvasSideLength,
+            this.canvasSideLength
+        );
         this.dataManager = new DataManager(
             this.dies,
             this.quadrant,
-            { width: 500, height: 500 },
+            { width: this.canvasSideLength, height: this.canvasSideLength },
```

Comment:
In the following PR I expressed a concern that sometimes the height and width are being accessed directly from the canvas element instead of being saved as part of the data manager. You may want to sync-up to align on a pattern for saving the canvasSideLength: https://github.com/ni/nimble/pull/923#discussion_r1061018645

Reply:
I agree - this will be further discussed. However, given the fact that this feature was temporarily de-prioritized due to a large bug inflow on a different product, we will only be able to address this sometime next week, or later